server/src/main/java/org/jboss/as/server/deployment/DeploymentHandlerUtil.java

```diff
@@ -450,4 +459,13 @@ private static String getFormattedFailureDescription(OperationContext context) {
         }
         return hash;
     }
+
+    private static Map<String, Set<Index>> getAnnotationIndexCache(OperationContext context) {
+        Map<String, Set<Index>> result = context.getAttachment(ANNOTATION_INDEX_CACHE);
+        if (result == null) {
+            result = new HashMap<>();
```

Comment:
I changed this a fair bit since you looked at this, basically to attach a single AnnotationIndexSupport here instead of what should be its internal map.

But I changed that internal map to a CHM; thanks.